crates/sui-types/src/crypto.rs

```diff
@@ -203,53 +183,77 @@ pub fn random_key_pairs(num: usize) -> Vec<KeyPair> {
 
 // TODO: get_key_pair() and get_key_pair_from_bytes() should return KeyPair only.
 // TODO: rename to random_key_pair
-pub fn get_key_pair() -> (SuiAddress, KeyPair) {
+pub fn get_key_pair<KP: KeypairTraits>() -> (SuiAddress, KP) {
     get_key_pair_from_rng(&mut OsRng)
 }
 
 /// Generate a keypair from the specified RNG (useful for testing with seedable rngs).
-pub fn get_key_pair_from_rng<R>(csprng: &mut R) -> (SuiAddress, KeyPair)
+pub fn get_key_pair_from_rng<KP: KeypairTraits, R>(csprng: &mut R) -> (SuiAddress, KP)
 where
     R: rand::CryptoRng + rand::RngCore,
 {
-    let kp = KeyPair::generate(csprng);
+    let kp = KP::generate(csprng);
     (kp.public().into(), kp)
 }
 
 // TODO: C-GETTER
-pub fn get_key_pair_from_bytes(bytes: &[u8]) -> SuiResult<(SuiAddress, KeyPair)> {
-    let priv_length = <KeyPair as narwhal_crypto::traits::KeyPair>::PrivKey::LENGTH;
-    let sk =
-        PrivateKey::from_bytes(&bytes[..priv_length]).map_err(|_| SuiError::InvalidPrivateKey)?;
-    let kp: KeyPair = sk.into();
+pub fn get_key_pair_from_bytes<KP: KeypairTraits>(bytes: &[u8]) -> SuiResult<(SuiAddress, KP)> {
+    let priv_length = <KP as KeypairTraits>::PrivKey::LENGTH;
+    let sk = <KP as KeypairTraits>::PrivKey::from_bytes(&bytes[..priv_length])
+        .map_err(|_| SuiError::InvalidPrivateKey)?;
+    let kp: KP = sk.into();
     if kp.public().as_ref() != &bytes[priv_length..] {
         return Err(SuiError::InvalidAddress);
     }
     Ok((kp.public().into(), kp))
 }
 
-// TODO: replace this with a byte interpretation based on multicodec
-pub const SUI_SIGNATURE_LENGTH: usize = PublicKey::LENGTH + AccountSignature::LENGTH;
+//
+// Account Signatures
+//
 
-#[serde_as]
-#[derive(Eq, PartialEq, Copy, Clone, Serialize, Deserialize, JsonSchema)]
-pub struct Signature(
-    #[schemars(with = "Base64")]
-    #[serde_as(as = "Readable<Base64, Bytes>")]
-    [u8; SUI_SIGNATURE_LENGTH],
-);
+// 1. Eddsa
+// 2. Ecdsa
+// Enums for Signatures
+const FLAG_LENGTH: usize = 1;
```

Comment:
To simplify this PR I'd suggest we pin it to one byte for now, so we don't need LENGTH anymore. Unlikely we'll ever surpass the number 256 in the foreseeable future.

Reply:
I see, but the alternative for this is to replace it with a magic number - wouldn't this work nicer?

Reply:
I think no, 1 byte would be enough (there is no live product with crypto stuff that this num has been surpassed yet (even for decades old software)